pkg/provisioner/ironic/testserver/redfish.go

```diff
@@ -0,0 +1,703 @@
+package testserver
+
+import (
+	"net/http"
+	"testing"
+)
+
+// RedfishMock is a test server that implements Redfish Virtual Media semantics
+type RedfishMock struct {
+	*MockServer
+}
+
+// NewRedfish builds a redfish virtual media mock server
+func NewRedfish(t *testing.T) *RedfishMock {
+	return &RedfishMock{
+		MockServer: NewTLS(t, "redfish"),
+	}
+}
+
+// Endpoint returns the URL to the server
+func (m *RedfishMock) Endpoint() string {
+	return m.MockServer.RawEndpoint()
+}
+
+// RootService configures the server with a valid response for /redfish/v1/
+func (m *RedfishMock) rootService() *RedfishMock {
+	m.ResponseWithCode("/redfish/v1/", `
+	{
```

Comment:
As a pretty general comment, it's ok to hard-code the answer as a string, but it could be practical for maintenance reasons to define the minimum amount of fields possible (just the ones required by the test).  A valid alternative it's to use a JSON object and fill just the required fields, especially if `gofish` already offers such types

Reply:
This method must be public and used directly in the test setup

Reply:
I will try to clean some of the fields

Reply:
Most of fields are optional in redfish, +1 to removing all unnecessary for us.

Reply:
Done